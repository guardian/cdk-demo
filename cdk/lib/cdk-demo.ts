import type { App } from "@aws-cdk/core";
import type { GuStackProps } from "@guardian/cdk/lib/constructs/core";
import { GuStack } from "@guardian/cdk/lib/constructs/core";
import { GuS3Bucket } from "@guardian/cdk/lib/constructs/s3";

export class CdkDemo extends GuStack {
  migratedFromCloudFormation = true;

  constructor(scope: App, id: string, props: GuStackProps) {
    super(scope, id, props);

    new GuS3Bucket(this, "bucket", {
      app: "cdk-demo",
      versioned: true,
      existingLogicalId: {
        logicalId: "MyFirstBucket",
        reason: "To illustrate 'adoption' of a resource into CDK.",
      },
    });
  }
}
