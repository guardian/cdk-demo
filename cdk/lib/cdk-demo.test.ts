import "@aws-cdk/assert/jest";
import { SynthUtils } from "@aws-cdk/assert";
import { App } from "@aws-cdk/core";
import { CdkDemo } from "./cdk-demo";

describe("The CdkDemo stack", () => {
  it("matches the snapshot", () => {
    const app = new App();
    const stack = new CdkDemo(app, "CdkDemo", { stack: "deploy-tools" });
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});
