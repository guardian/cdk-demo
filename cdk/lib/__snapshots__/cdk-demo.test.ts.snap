// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CdkDemo stack matches the snapshot 1`] = `
Object {
  "Description": "Demo app",
  "Parameters": Object {
    "Stage": Object {
      "AllowedValues": Array [
        "CODE",
        "PROD",
      ],
      "Default": "CODE",
      "Description": "Stage name",
      "Type": "String",
    },
  },
  "Resources": Object {
    "MyFirstBucket": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/cdk-demo",
          },
          Object {
            "Key": "Stack",
            "Value": "deploy-tools",
          },
          Object {
            "Key": "Stage",
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
        "VersioningConfiguration": Object {
          "Status": "Enabled",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;
